Pulling data (#1373)
ihmeuw/vivarium_research: docs/source/model_design/vivarium_overview/data_sources/index.rst

.. _data:

===================================
Understanding and Pulling GBD Data
===================================

> Global Burden of Disease (GBD) Study data is a primary data source for our simulation models. 

Reviewer: ```suggestion
Global Burden of Disease (GBD) Study data is a fundamental data source for our simulation models. 
```

There is some term-of-art about "primary" vs "secondary" data and I thnk you just mean "very important"

Author: Yes, agreed, thanks for catching!

Understanding what data is available in the GBD and what modeling processes produced it is a 
difficult task. Some helpful resources for understanding the GBD study are listed below: 

- IHME onboarding trainings

> - GBD capstone papers and their methods appendices 

Reviewer 2: I think adding links to 2019 capstone papers would be helpful here!

- `The GBD compare tool <https://vizhub.healthdata.org/gbd-compare/>`_, which allows you to visualize GBD estimates
- `GBD ID viewer <https://shiny.ihme.washington.edu/connect/#/apps/1cbf4a06-bfe9-4896-9028-9f4d23cc75c6/access>`_
- `GBD risk factors toolbox <https://shiny.ihme.washington.edu/content/13/>`_
- Your simulation science team members!
- Talking to GBD modelers directly

Pulling GBD Data using *Shared Functions*
-----------------------------------------

IHME central computation maintains functions for accessing GBD data, referred to as "Shared Functions." `The main HUB page for shared functions can be found here <https://hub.ihme.washington.edu/display/SF/Shared+Functions+Home>`_

> Note that there is a central computation maintained conda environment that is guarenteed to have the latest version of all GBD shared functions, called :code:`gbd_env`, as described on the Shared Functions HUB page. 

Reviewer 2: ```suggestion
Note that there is a central computation maintained conda environment that is guaranteed to have the latest version of all GBD shared functions, called :code:`gbd_env`, as described on the Shared Functions HUB page. 
```

- Note that archived GBD rounds (for example, GBD 2017) may require archived GBD environments to access - see the "Current and Archive GBD environments" subpage for more details.

> - Also note that while the :code:`gbd_env` environment is guarenteed to have the most up to date versions of shared functions, it is unlikely to support additional functions you may want to use, which is a downside of using this environment.

Reviewer 2: ```suggestion
- Also note that while the :code:`gbd_env` environment is guaranteed to have the most up to date versions of shared functions, it is unlikely to include additional packages you may want to use, which is a downside of using this environment.
```

   *  - :code:`exposure_sd`
      - Risk exposure standard deviation
      - rei_id
      - Only for continuous risks
   *  - :code:`rr`
      - Risk factor relative risk
      - rei_id, cause_id
      - Will return values for all affected causes unless a cause_id is specified
   *  - :code:`burdenator`
      - Risk attributable burden (deaths/dalys/ylls/ylds) and mediated/aggregated PAFs
      - cause_id, rei_id
      - 
   *  - :code:`paf`
      - Pre-burdenator (non-finalized) PAF estimates
      - rei_id, cause_id
      - 
   *  - :code:`sev`
      - Summary exposure values
      - rei_id
      - 
   *  - :code:`tmrel`
      - Risk factor theoretical minimum exposure level
      - rei_id
      - 
   *  - :code:`rr_max`
      - Relative risk maximum value
      - rei_id, cause_id
      - 
   *  - :code:`codem`
      - codem models and custom cod models
      - cause_id
      - 
   *  - :code:`stgpr`
      - ST-GPR models
      - modelable_entity_id
      - If you pass an MEID with a dismod model type but try to use the ST-GPR source, get_draws will use the epi source instead.

Handling GBD versioning
++++++++++++++++++++++++

> Decomposition (or "decomp") steps are a versioning scheme used in some GBD rounds that allowed updates to GBD results based on iterative updates to certain parts of the computation process. For instance, the first step may be equivalent to the prior GBD round in all aspects except for an updated demographic model; the second step may be equivalent to the prior steps, but with updated risk exposures; and so on. This process allowed GBD researchers to evaluate individual components of the many changes included in a GBD round advancement influenced the main results of the GBD study rather than updating the entire pipeline at once. 

Reviewer 2: ```suggestion
Decomposition (or "decomp") steps are a versioning scheme used in some GBD rounds that allowed updates to GBD results based on iterative updates to certain parts of the computation process. For instance, the first step may be equivalent to the prior GBD round in all aspects except for an updated demographic model; the second step may be equivalent to the prior steps, but with updated risk exposures; and so on. This process allowed GBD researchers to evaluate how individual components of the many changes included in a GBD round advancement influenced the main results of the GBD study, rather than updating the entire pipeline at once.
```

I think this sentence is missing a word, but I'm not sure if this is the right one :)
